docs/how-tos/adding-event-bus-support-to-an-event.rst

```diff
@@ -0,0 +1,213 @@
+
+- The event is only used within the same application process and cannot be scoped to other services.
+- The :term:`Event Payload` contains data types that are not supported by the event bus, and it is not possible to refactor the :term:`Event Payload` to use supported data types.
+
+When adding support is not possible do the following:
+
+- Add it to the ``KNOWN_UNSERIALIZABLE_SIGNALS`` list in the ``openedx_events/tooling.py`` file so the event bus ignores it.
+- Add a ``warning`` in the event's docstring to inform developers that the event is not compatible with the event bus.
+
+If you don't add the event to the ``KNOWN_UNSERIALIZABLE_SIGNALS`` list, the CI/CD pipeline will fail for the missing Avro schema that could not be generated for the :term:`Event Payload`. If you don't add a warning in the event's docstring, developers might try to send the event across services and encounter issues.
+
+Step 2: Define the Event Payload
+--------------------------------
+
+An Open edX Event is compatible with the event bus when its payload can be serialized, sent, and deserialized by other services. The payload, structured as `attrs data classes`_, must align with the event bus schema format which in this case is the :term:`Avro Schema`. This schema is used to serialize and deserialize the :term:`Event Payload` when sending it across services.
+
+This ensures the event can be sent by the producer and be then re-emitted by the same instance of `OpenEdxPublicSignal`_ on the consumer side. For more information on the event bus schema format, refer to the :doc:`../decisions/0004-external-event-bus-and-django-signal-events` and :doc:`../decisions/0005-external-event-schema-format` decision records.
+
+Here is an example of an :term:`Event Payload` structured as `attrs data classes`_ that align with the event bus schema format:
+
+.. code-block:: python
+
+    @attr.s(frozen=True)
+    class UserNonPersonalData:
+        """
+        Attributes defined for Open edX user object based on non-PII data.
+
+        Arguments:
+            id (int): unique identifier for the Django User object.
+            is_active (bool): indicates whether the user is active.
+        """
+
+        id = attr.ib(type=int)
+        is_active = attr.ib(type=bool)
+
+    @attr.s(frozen=True)
+    class UserPersonalData:
+        """
+        Attributes defined for Open edX user object based on PII data.
+
+        Arguments:
+            username (str): username associated with the Open edX user.
+            email (str): email associated with the Open edX user.
+            name (str): name associated with the Open edX user's profile.
+        """
+
+        username = attr.ib(type=str)
+        email = attr.ib(type=str)
+        name = attr.ib(type=str, factory=str)
```

Comment:
Not sure if `factory` here is confusing matters or something you want to describe

Reply:
I dropped the example here: https://github.com/openedx/openedx-events/pull/428/commits/51683fd1adca9ef6a9864b56dedc226aeb2da903#diff-67886caf4b3357c606ceb6d3ea25e3839b6056f13b22d48a146431adc0fa829dR53